src/backend/access/appendonly/aosegfiles.c

```diff
@@ -1506,7 +1513,7 @@ gp_update_ao_master_stats(PG_FUNCTION_ARGS)
 	int64		result;
 	Snapshot	appendOnlyMetaDataSnapshot;
 
-	Assert(Gp_role == GP_ROLE_DISPATCH);
+	VALIDATE_GP_ROLE();
```

Comment:
`;` is extra for macro.

Reply:
> ; is extra for macro.

I would prefer to remove the `;` in the macro at the end of the loop statement. That would make the code, where the macro is used, more consistent.